Update 14-objects.mkd

jdougnac opened the pull request:
Traducido el capítulo, desde cero, ya que por lo visto no existía en la edición anterior del libro en español.
 Dudas:
L. 18: Collection Data Structures = ¿Estructuras de datos recolectados?
L. 120: Input = ¿Dato de entrada?
Dot operator = ¿Operador de punto?
En la línea 282, no entiendo a qué se refiere con object pointer y la referencia a 'self' en el paréntesis, por lo que 
ruego revisar, pues es posible que me haya equivocado en la traducción.
L. 326: Object lifecycle = ¿Ciclo de vida del objeto?

jdougnac committed:
Update 14-objects.mkd

Traducido el capítulo, desde cero, ya que por lo visto no existía en la edición anterior del libro en español.
 Dudas:
L. 18: Collection Data Structures = ¿Estructuras de datos recolectados?
L. 120: Input = ¿Dato de entrada?
Dot operator = ¿Operador de punto?
En la línea 282, no entiendo a qué se refiere con object pointer y la referencia a 'self' en el paréntesis, por lo que 
ruego revisar, pues es posible que me haya equivocado en la traducción.
L. 326: Object lifecycle = ¿Ciclo de vida del objeto?

jdougnac commented:
@don-charlie-pc  , etiquetándote tal como me pediste.

@csev tagging our benevolent overlord as well, as a sign of respect

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -1,97 +1,100 @@
-
-Object-oriented programming
+Programación Orientada a Objetos
 ===========================
```
For the lines under titles or sections (like line 5, below, and all other sections of the chapter) please insert characters according to the length of its title, like:

```
Programación Orientada a Objetos
================================
```

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -1,97 +1,100 @@
+* Código secuencial
+* Código condicional (declaraciones 'if')
+* Código repetitivo (bucles)
+* Almacenar y reutilizar (functiones)
```
functiones -> funciones

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -1,97 +1,100 @@
-while ignoring the other 999,950 lines of code for the moment.
+En cierto modo, la programación orientada a objetos es una forma
+de ordenar tu código de tal manera que puedas enfocarte en 50
+líneas de código y entenderlas, e ignorar las otras 999.950
```
Let's keep the comma here, 999.950 -> 999,950

jdougnac replied:
Ok, pero en castellano el divisor de números es el punto. Si dejamos la coma, estaría significando "novecientas noventa y nueve coma noventa y cinco", no "novecientas noventa y nueve mil novecientas cincuenta".

don-charlie-pc replied:
De acuerdo, entonces hay que mantener el formato con el punto.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -1,97 +1,100 @@
-You should approach this chapter as a way to learn some terms and concepts and work
-through a few simple examples to lay a foundation for future learning.
+Al igual que muchos aspectos de la programación, es necesario aprender
+los conceptos de la programación orientada objetos para poder utilizarlos
```
"a" missing -> ...programación orientada **a** objetos...

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -1,97 +1,100 @@
-The first line *constructs* an object of type `list`, the second and third
-lines *call* the `append()` *method*, the fourth line calls
-the `sort()` method, and the fifth line *retrieves* the item at position 0.
+La primera línea *construye* un objeto de tipo 'list', la segunda y tercera
```
In this specific case, use backticks instead of single quotes, as it will render as a code-block. Same for the "sort()" word in line 72, "append()" on line 71 (add the parenthesis), and __getitem__() in line 74.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -1,97 +1,100 @@
-In this code, we call the `__getitem__` method in the `list` class
-and *pass* the list and the item we want retrieved from the list
-as parameters.
+En este programa, llamamos al método '__getitem__' en la clase 'lista'
```
When the original text has backticks, let's keep them. Sorry if I didn't explain myself on the comment about quotes and backticks on the previous review. Since this is markdown, the backticks `` are used to indicate a code statement, which renders as `this`. Just keep the format as it comes from the original chapter. Thanks, and sorry for troubling you!

jdougnac replied:
No me aproblema corregir errores referidos a la traducción, ni a los typos, etc. Sin embargo, recuerdo que para el capítulo anterior que traduje me dijiste específicamente que los backticks los cambiara por '' en todos los casos.

 El no tener claridad sobre qué es correcto y qué no me entorpece bastante la labor de traducción. Sería útil contar con una guía, aunque fuera un documento corto, para evitar estos problemas a futuro.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -1,97 +1,100 @@
-We can take a look at the capabilities of an object by
-looking at the output of the `dir()` function:
+Podemos ver las capacidades de un objeto mirando
+el resultado de la función 'dir()':
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -107,123 +110,121 @@ looking at the output of the `dir()` function:
-We call the `tags()` method on the returned object that returns a dictionary of tag
-objects. We loop through the tags and call the `get()` method for each tag
-to print out the `href` attribute.
+Convertimos la URL en una cadena y luego pasamos a ésta por 'urllib' para
```
Use backticks here, for the urllib and socket words in this paragraph. Also for html.parser in line 152.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -107,123 +110,121 @@ looking at the output of the `dir()` function:
+para llevar a cabo la conexión que recupera los datos. Tomamos la cadena
+que retorna 'urllib' y se la entregamos a BeautifulSoup para su análisis.
+BeautifulSoup utiliza el objeto 'html.parser'^[https://docs.python.org/3/library/html.parser.html]
+y retorna un objeto. Luego, llamamos al método 'tags()' en el objeto retornado, lo que
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -107,123 +110,121 @@ looking at the output of the `dir()` function:
+BeautifulSoup utiliza el objeto 'html.parser'^[https://docs.python.org/3/library/html.parser.html]
+y retorna un objeto. Luego, llamamos al método 'tags()' en el objeto retornado, lo que
+retorna un diccionario de etiquetas. Nos desplazamos por las etiquetas y llamamos el 
+método 'get()' por cada etiqueta para imprimir el atributo 'href'.
```
Use backticks here for get() and href.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -107,123 +110,121 @@ looking at the output of the `dir()` function:
-solve and ignore the other parts of the program.
+Una de las ventajas del enfoque orientado a objetos es que puede ocultar la
+complejidad de un programa. Por ejemplo, aunque necesitamos saber cómo usar
+el código de 'urllib' y BeautifulSoup, no necesitamos saber cómo funcionan
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -107,123 +110,121 @@ looking at the output of the `dir()` function:
-in each object of type `PartyAnimal`.  The class is like a cookie cutter
-and the objects created using the class are the
-cookies^[Cookie image copyright CC-BY
+Cada método parece una función: comienzan con la palabra clave 'def' y
```
Use backticks here where applicable, same for the next paragraph.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
-In a way this is quite similar to the following line which we have
-been using all along:
+Es aquí que le ordenamos a Python construir (es decir, crear) un *objeto* o
+*instancia* de la clase 'PartyAnimal'. Se ve como si fuera una llamada de función
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
+Es aquí que le ordenamos a Python construir (es decir, crear) un *objeto* o
+*instancia* de la clase 'PartyAnimal'. Se ve como si fuera una llamada de función
+a la clase misma. Python construye el objeto con los datos y métodos adecuados,
+asignándolo luego a la variable 'an'. En cierto modo, esto es muy similar
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
-Here we instruct Python to construct an object using the `dict` template
-(already present in Python), return the instance of dictionary,
-and assign it to the variable `counts`.
+Aquí le ordenamos a Python construir un objeto usando la plantilla 'dict'
```
Use backticks here for dict and counts on line 250.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
-When the `PartyAnimal` class is used to construct an object, the variable `an`
-is used to point to that object.  We use `an` to access the code and data for that
-particular instance of the `PartyAnimal` class.
+Cuando se usa la clase 'PartyAnimal' para construir un objeto, la variable 'an'
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
-When the `party` method is called, the first parameter (which we call by convention `self`)
-points to the particular instance of the PartyAnimal object that `party` is called from.
-Within the `party` method, we see the line:
+Al llamar al método 'party', el primer parámetro (al que por convención llamamos 'self')
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
 
-The following line is another way to call the `party` method within the `an`
-object:
+La siguiente línea muestra otra manera de llamar al método 'party' dentro del 
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
-pass the object pointer `an` as the first parameter (i.e., `self` within the method).
-You can think of `an.party()` as shorthand for the above line.
+En esta variante, accedemos al código desde el interior de la clase y explícitamente
+pasamos el objeto 'an' como el primer parámetro (es decir, 'self' dentro del método).
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -291,22 +292,23 @@ So far 3
 
-The object is constructed, and the `party` method is called four times, both
-incrementing and printing the value for `x` within the `an` object.
+El objeto es construido, y el método 'party' es llamado cuatro veces,
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -291,22 +292,23 @@ So far 3
-`dir` function to examine the capabilities of a variable.  We can also use `type` and
-`dir` with the classes that we create.
+Como hemos visto, en Python todas las variables tienen un tipo. Podemos usar
+la función 'dir' incluida en Python para examinar las características de una variable.
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -291,22 +292,23 @@ So far 3
-`dir` with the classes that we create.
+Como hemos visto, en Python todas las variables tienen un tipo. Podemos usar
+la función 'dir' incluida en Python para examinar las características de una variable.
+También podemos usar 'type' y 'dir' con las clases que creemos.
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -317,31 +319,31 @@ Type <class 'int'>
-You can see that using the `class` keyword, we have created a new type.
-From the `dir` output, you can see both the `x` integer attribute and the `party`
-method are available in the object.
+Puedes ver que, usando la palabra clave 'class', hemos creado un nuevo tipo.
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -317,31 +319,31 @@ Type <class 'int'>
+como el método 'party' están disponibles dentro del objeto.
 
-Object lifecycle
+Ciclo de vida del objeto
```
Quizá suena mejor "Ciclo de Vida de un Objeto"

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -351,56 +353,56 @@ I am destructed 2
-As Python constructs our object, it calls our `__init__` method to
-give us a chance to set up some default or initial values for the object.
-When Python encounters the line:
+Cuando Python construye el objeto, llama a nuestro método '__init__' para
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -351,56 +353,56 @@ I am destructed 2
-
-Multiple instances
+efectivamente "tira a la basura" el objeto para reutilizar la variable
+'an', almacenando el valor '42'. Justo en el momento en que nuestro objeto
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -351,56 +353,56 @@ I am destructed 2
-The constructor has both a `self` parameter that points to the object instance
-and additional parameters that are passed into the constructor as the
-object is constructed:
+El constructor tiene tanto un parámetro 'self', que apunta a la instancia
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -351,56 +353,56 @@ I am destructed 2
 
-Within the constructor, the second line copies the parameter (`nam`) that is
-passed into the `name` attribute within the object instance.
+Dentro del constructor, la segunda línea copia el parámetro ('nam'), el que
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -351,56 +353,56 @@ I am destructed 2
 
-The output of the program shows that each of the objects (`s` and `j`) contain their own
-independent copies of `x` and `nam`:
+El resultado del programa muestra que cada objeto ('s' y 'j') contienen sus propias
```
Use backticks here.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -410,28 +412,28 @@ Jim party count 1
-(`party`) from the `PartyAnimal` class are *inherited* by the `CricketFan` class.
-For example, within the `six` method in the `CricketFan` class, we 
-call the `party` method from the `PartyAnimal` class.
+Cuando definimos la clase 'CricketFan', indicamos que estamos extendiendo
```
Use backticks in this paragraph and the one below.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -444,112 +446,107 @@ Jim points 6
-you might want to explore a few Computer Science courses.]
-
-Glossary
+La primera línea construye un *objeto* de clase 'list'. Cuando Python crea el objeto
```
Use backticks where applicable.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -1,97 +1,100 @@
-In later chapters, we explored simple variables as well as collection
-data structures like lists, tuples, and dictionaries.
+En capítulos posteriores, exploramos las variables simples, así como 
+estructuras de datos recolectados, tales como listas, tuplas y diccionarios.
```
I think it sounds better: "estructuras de datos de colecciones"

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -107,123 +110,121 @@ looking at the output of the `dir()` function:
-A program in its most basic form takes some input, does some processing, and
-produces some output.  Our elevator conversion program demonstrates a very
-short but complete program showing all three of these steps.
+En su forma más básica, un programa toma algún dato de entrada, lo procesa
```
Sounds good!

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
 
-This syntax using the *dot* operator is saying 'the x within self.' Each time `party()` is
-called, the internal `x` value is incremented by 1 and the value is printed out.
+Esta sintaxis utiliza el operador de *punto*, con lo que significa 'la x dentro de self'.
```
Sounds good.

don-charlie-pc commented on book3/14-objects.mkd:
```diff
@@ -234,55 +235,55 @@ an = PartyAnimal()
-pass the object pointer `an` as the first parameter (i.e., `self` within the method).
-You can think of `an.party()` as shorthand for the above line.
+En esta variante, accedemos al código desde el interior de la clase y explícitamente
+pasamos el objeto 'an' como el primer parámetro (es decir, 'self' dentro del método).
```
Update to "pasamos el apuntador del objeto \`an\`"

jdougnac replied:
acabo de terminar de efectuar las correcciones mencionadas aquí. Usé buscar y reemplazar para cambiar los '' por backticks en todo el documento.

don-charlie-pc reviewed (commented):
Gracias @jdougnac por la traducción entera del capítulo!
Tengo un par de comentarios menores, la mayoría relacionados al uso de los backticks y las comillas. Básicamente, hay que utilizarlas tal como están en el texto original, una disculpa si te provoqué confusión sobre el uso de las comillas en el review anterior. Los backticks se utilizan para indicar código, de modo que Markdown hace un render diferente para ese texto. Gracias de nuevo!

jdougnac committed:
Update 14-objects.mkd

Efectuados todos los cambios mencionados en los comentarios. Usé buscar y reemplazar para cambiar todos los '' por backticks.

don-charlie-pc commented:
Gracias @jdougnac . Así como mencionas, sería bueno tener una referencia o guía más detallada sobre la traducción en cuanto a caracteres especiales. También soy relativamente nuevo en Markdown y llevo colaborando en este proyecto desde hace un par de meses. Aquí hay una buena referencia para Markdown https://www.markdownguide.org/basic-syntax/ 
Mil disculpas por las molestias que te ocasioné con las confusiones sobre las comillas.

jdougnac commented:
@don-charlie-pc no hay problema. Guardaré la guía a fin de usarla como orientación para el próximo capítulo.

 ¿Sigo con el capítulo 16?

don-charlie-pc commented:
Suena bien! Adelante, muchas gracias!
